### ISSUE TITLE
php-dev broken when upgrading from Ubuntu 19.10 to 20.04

### ISSUE THREAD

**Issue by Zenexer** [oerdnj/deb.sury.org#1408]
1. php-dev depends on php7.4-dev
2. php7.4-dev depends on libpcre2-dev (>= 10.30)
3. libpcre2-dev depends on libpcre2-8-0 (= 10.34-7)

```
The following packages have unmet dependencies:
 libpcre2-dev : Depends: libpcre2-8-0 (= 10.34-7) but 10.34-7+ubuntu19.10.1+deb.sury.org+1 is to be installed
```

```
% cat /etc/apt/sources.list.d/ondrej-ubuntu-php.list
deb http://ppa.launchpad.net/ondrej/php/ubuntu focal main
deb-src http://ppa.launchpad.net/ondrej/php/ubuntu focal main

% lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04 LTS
Release:        20.04
Codename:       focal
```

The problem appears to be that `10.34-7+ubuntu19.10.1+deb.sury.org+1` is considered newer than `10.34-7`, so the `10.34-7` from the main repo won't be installed when upgrading from Ubuntu 19.10 to 20.04, even though `10.34-7+ubuntu19.10.1+deb.sury.org+1` isn't in the 20.04 sury.org repo.

Manual workaround: `sudo apt install libpcre2-8-0=10.34-7`  Ignore the warnings about downgrading.

**Comment by oerdnj** [comment]
pcre2 has been updated to 10.35 and uploaded to focal, it's building now.

**Comment by jult** [comment]
This is still broken, but then for php7.4 on Debian Buster (10).
Can't get php7.4-dev to install.

```
The following packages have unmet dependencies:
 libpcre2-dev : Depends: libpcre2-8-0 (= 10.32-5) but 10.34-7+0~20191220.5+debian10~1.gbp555bb9 is to be installed
                Depends: libpcre2-16-0 (= 10.32-5) but 10.34-7+0~20191220.5+debian10~1.gbp555bb9 is to be installed
                Depends: libpcre2-32-0 (= 10.32-5) but 10.34-7+0~20191220.5+debian10~1.gbp555bb9 is to be installed
E: Unable to correct problems, you have held broken packages.

```

**Comment by wapmorgan** [comment]
It's updated now with versions, but still has an issue in requirements.
Can be fixed in ubuntu 20.04 now:
`sudo apt install libpcre2-16-0=10.34-7 libssl1.1=1.1.1f-1ubuntu2.2`

**Comment by tvvpm** [comment]
I'm trying to install php7.4-dev to compile pdo oci extension. I'm getting the same problem.

The following packages have unmet dependencies:
 libpcre2-dev : Depends: libpcre2-8-0 (= 10.36-2+deb11u1) but 10.40-1+0~20220713.16+debian11~1.gbpb6cec5 is to be installed
E: Unable to correct problems, you have held broken packages
